src/Raven.Client/Document/DocumentConvention.cs

```diff
@@ -101,13 +102,26 @@ public DocumentConvention()
             MaxLengthOfQueryUsingGetUrl = 1024 + 512;
             ApplyReduceFunction = DefaultApplyReduceFunction;
             ReplicationInformerFactory = (url, jsonRequestFactory) => new ReplicationInformer(this, jsonRequestFactory);
-            CustomizeJsonSerializer = serializer => { };
+            CustomizeJsonSerializer = serializer => { }; // todo: remove this or merge with SerializeEntityToJson
             FindIdValuePartForValueTypeConversion = (entity, id) => id.Split(new[] { IdentityPartsSeparator }, StringSplitOptions.RemoveEmptyEntries).Last();
             ShouldAggressiveCacheTrackChanges = true;
             ShouldSaveChangesForceAggressiveCacheCheck = true;
             IndexAndTransformerReplicationMode = IndexAndTransformerReplicationMode.Indexes | IndexAndTransformerReplicationMode.Transformers;
             AcceptGzipContent = true;
             RequestTimeThresholdInMilliseconds = 100;
+
+            SerializeEntityToJson  = (object entity, out Stream jsonStream) =>
```

Comment:
Instead of allocating the stream on each entity, allocate it once, and pass it to the delegate.